### ISSUE TITLE
Cast nested `List` to `LargeList`

### ISSUE THREAD

**Issue by Weijun-H** [apache/arrow-datafusion#8305]
### Is your feature request related to a problem or challenge?

datafusion cannot cast nested `List` to `LargeList`, like `List(List(Int64))` to `LargeList(LargeList(Int64))`.
```
select arrow_cast(make_array(make_array(1, 2, 3), make_array(4, 5, 6)), 'LargeList(LargeList(Int64))');

Error during planning: Cannot automatically convert List(Field { name: "item", data_type: List(Field { name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }) to LargeList(Field { name: "item", data_type: LargeList(Field { name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} })
```

### Describe the solution you'd like

https://github.com/apache/arrow-rs/blob/master/arrow-cast/src/cast.rs#L133-L135
```
        (List(list_from), LargeList(list_to)) => {
            list_from.data_type() == list_to.data_type()
        }
```

**Comment by Weijun-H** [comment]
Will be fixed by arrow-rs 50.0

**Comment by alamb** [comment]
> Will be fixed by arrow-rs 50.0

This is tracked by https://github.com/apache/arrow-rs/issues/5234
